apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts

```diff
@@ -54,11 +56,53 @@ export function StudentAssessmentInstance({
           navPage: 'assessment_instance',
         })}
         <main id="content" class="container">
+          ${canDeleteAssessmentInstance
+            ? html`
+                <div class="alert alert-warning d-flex align-items-center">
+                  <span class="mr-3">
+                    Additional actions are available because you are viewing your own assessment
+                    instance and have student data editor permissions.
+                  </span>
+
+                  <form method="POST" class="ml-auto mr-3">
+                    <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+                    <div class="dropdown">
+                      <button
+                        class="btn btn-warning dropdown-toggle"
+                        type="button"
+                        data-toggle="dropdown"
+                        aria-expanded="false"
+                      >
+                        Actions
+                      </button>
+
+                      <div class="dropdown-menu dropdown-menu-right">
+                        <button
+                          type="submit"
+                          class="dropdown-item"
+                          name="__action"
+                          value="delete_instance"
+                        >
+                          <i class="fas fa-arrows-rotate"></i>
+                          Regenerate assessment instance
```

Comment:
TODO: if we like calling this "regenerate" instead of "delete", update the word "delete" elsewhere as needed.

Reply:
Since this was moved to a dropdown, can we have both? The only difference in that case is where it redirects to after deleting the instance.

Reply:
I added these options in 408d7fe1c - I agree that we don't need to change delete in the functions, this is just special handling on what we do after the delete.